test/extended/operators/olm.go

```diff
@@ -151,6 +151,38 @@ var _ = g.Describe("[sig-operator] OLM should", func() {
 	})
 })
 
+func hasRedHatOperatorsSource(oc *exutil.CLI) error {
+	spec, err := oc.AsAdmin().Run("get").Args("operatorhub/cluster", "-o=jsonpath={.spec}").Output()
+	if err != nil {
+		return fmt.Errorf("Error reading operatorhub spec: %s", spec)
+	}
+	type Source struct {
+		Name     string `json:"name"`
+		Disabled bool   `json:"disabled"`
+	}
+	type Spec struct {
+		DisableAllDefaultSources bool     `json:"disableAllDefaultSources"`
+		Sources                  []Source `json:"sources"`
+	}
+	parsed := Spec{}
+	err = json.Unmarshal([]byte(spec), &parsed)
+	if err != nil {
+		return fmt.Errorf("Error unmarshalling operatorhub spec: %s", spec)
```

Comment:
This changes the behavior of the test - before you were reporting errors as test failures, now you are reporting them as skips.  It would always be better to report errors as test failures, and skip is a boolean with a hardcoded message.  You don't need to return the message from this function, just inline the message in the test block.

We don't want to suddenly start failing unmarshal (because of a schema change) and accidentally always skip this test

Reply:
Oh, I see. Reworked this to return bool and err, which is expected to be nil on OCP and OKD